src/app/projects/project-list-filters/project-list-filters.component.ts

```diff
@@ -0,0 +1,153 @@
+import { CommentPeriodService } from 'app/services/commentperiod.service';
+import { ProjectType } from 'app/models/project';
+import { TableParamsObject } from 'app/shared/components/table-template/table-params-object';
+
+export interface UpdateMatchingData {
+  source: string;
+  activeFilters: string[];
+}
+
+export interface ProjectTypeFullNameMap {
+  lup: string;
+  flp: string;
+  wpag: string;
+}
+
+@Component({
+  selector: 'app-project-list-filters',
+  templateUrl: './project-list-filters.component.html',
+  styleUrls: ['./project-list-filters.component.scss']
+})
+
+export class ProjectListFiltersComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() projects: Array<Project> = []; // from project list component
+  @Input() tableParams: TableParamsObject = new TableParamsObject(); // from project list component
+  @Output() updateMatching = new EventEmitter<UpdateMatchingData>(); // to project list component
+
+  public loading: boolean = false;
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+  public applyFilterCount: number = 0;
+
+	public projectTypeFilters: ProjectType[] = [
+		{name: 'Land Use Planning', checked: true},
+		{name: 'Forest Landscape Planning', checked: true},
+		{name: 'Water Planning and Governance', checked: true}
+	];
+
+  public fullNames: ProjectTypeFullNameMap = {
+    lup: 'Land Use Planning',
+    flp: 'Forest Landscape Planning',
+    wpag: 'Water Planning and Governance',
+  }
+
+  constructor(
+    public commentPeriodService: CommentPeriodService, // also used in template
+  ) {}
+
+  // Apply all filters and emit data to parent component
+  public ngOnInit() {
+    this.internalApplyAllFilters();
+  }
+
+  // Triggered when the change detection is triggered from the parent component.
+  public ngOnChanges() {
+    this.internalApplyAllFilters('search');
+  }
+
+  /**
+   * Applies the current this.tableParams values to the filter controls.
+   */
+  public applyParamsToFilterControls() {
+    if (this.tableParams.projectTypes) {
+      const projectTypesArray = this.tableParams.projectTypes.split(',');
+      const expandedProjectTypesArray = projectTypesArray.map(item => this.fullNames[item] || item);
+      this.projectTypeFilters.forEach(ptf => {
+        ptf.checked = expandedProjectTypesArray.includes(ptf.name) || 'all' === this.tableParams.projectTypes ? true : false;
+      })
+    }
+    this.applyFilterCount++;
+  }
+
+	/**
+	 * Handles an event change to the project type filter
+	 * 
+	 * @param {MatCheckboxChange} event The event that is passed from the project type filter checkbox
+	 * @returns {void}
+	 */
+	public handleProjectTypeChange(event: MatCheckboxChange): void {
+		const index = this.projectTypeFilters.map(filter => filter.name).indexOf(event.source.value);
+		if (-1 !== index) {
+			this.projectTypeFilters[index].checked = event.checked;
+			this.internalApplyAllFilters('project type filter');
+		}
+	}
+
+  /**
+   * Applies filters.
```

Comment:
```suggestion
   * Applies filters by getting those that are selected, passing them to the shared table component, and then emitting an event so that the table is updated.
```

Reply:
That's fair, since we're in the spirit of thorough explanations.